src/mbed_tools/project/mbed_program.py

```diff
@@ -63,7 +61,7 @@ def from_url(cls, url: str, dst_path: Path, check_mbed_os: bool = True) -> "Mbed
                 "to an empty directory."
             )
         logger.info(f"Cloning Mbed program from URL '{url}'.")
-        repo = git_utils.clone(url, dst_path)
+        git_utils.clone(url, dst_path)
```

Comment:
Do we have any tests covering where this clone fails? Maybe the `feature-cmake` branch or whatever reference in the `mbed-os.lib` file is broken. I know we have a test `test_raises_when_clone_fails()` that checks an exception is thrown from `clone()`, but I'm not sure we have anything testing how we handle invalid references in mocked mbed-os.lib files.

Reply:
At the moment we just propagate any error thrown from git. We don't do any extra checking of the lib references. We probably should. I'll see if I can add something.

Reply:
If we just let git deal with invalid references we get the following error message:

```
$ mbedtools init ../../programs/mytestprog
Creating a new Mbed program at path '../../programs/mytestprog'.
Downloading mbed-os and adding it to the project.
ERROR: Failed to check out revision 'featurhakhkahkah'. Error from VCS:
  stderr: 'error: pathspec 'featurhakhkahkah' did not match any file(s) known to git'
```

This seems OK to me, do you think we need to do anything more here? Maybe we could add which `.lib` file is corrupted to the error message?

Reply:
That error looks fine. Should we have a test to ensure that the error from git is propagated as expected?

Reply:
We do have a test to ensure we're propagating the error here: https://github.com/ARMmbed/mbed-tools/blob/8800c3d07a7e31bb1c477eebe88fb68887b873cd/tests/project/_internal/test_git_utils.py#L64